lib/sequel/plugins/forbid_lazy_load.rb

```diff
@@ -179,7 +192,7 @@ def _load_associated_objects(opts, dynamic_opts=OPTS)
         end
       end
 
-      module DatasetMethods
+      module ForbidByDefault
```

Comment:
Renamed as per https://github.com/jeremyevans/sequel/discussions/2286#discussioncomment-12512504, but should we maybe append "DatasetMethods" to the end of this to make it clear that these are specifically for that?